src/cell.h

```diff
@@ -300,20 +300,20 @@ struct Cell {
         doc->AddUndo(this);
     }
 
-    void Save(wxDataOutputStream &dos) const {
+    void Save(wxDataOutputStream &dos, Cell *ocs) const {
         dos.Write8(celltype);
         dos.Write32(cellcolor);
         dos.Write32(textcolor);
         dos.Write8(drawstyle);
         if (HasTextState()) {
-            dos.Write8(grid ? TS_BOTH : TS_TEXT);
+            dos.Write8(grid ? ((this == ocs) ? TS_BOTH | TS_SELECTION_MASK : TS_BOTH) : ((this == ocs) ? TS_TEXT | TS_SELECTION_MASK : TS_TEXT));

```

Comment:
why do this twice? just create the flags first in a var, then `|` if selected.

Reply:
I revised the commit. Is this ok now?

Reply:
I revised this. Is this ok now?